src/Widgets/EndSessionDialog.vala

```diff
@@ -79,35 +72,39 @@ public class Session.Widgets.EndSessionDialog : Gtk.Dialog {
                 break;
         }
 
-        set_position (Gtk.WindowPosition.CENTER_ALWAYS);
-        set_keep_above (true);
-        stick ();
-
         var image = new Gtk.Image.from_icon_name (icon_name, Gtk.IconSize.DIALOG);
         image.valign = Gtk.Align.START;
 
-        var heading = new Gtk.Label (heading_text);
-        heading.get_style_context ().add_class (Granite.STYLE_CLASS_PRIMARY_LABEL);
-        heading.xalign = 0;
+        var primary_label = new Gtk.Label (heading_text);
+        primary_label.get_style_context ().add_class (Granite.STYLE_CLASS_PRIMARY_LABEL);
+        primary_label.max_width_chars = 50;
+        primary_label.wrap = true;
+        primary_label.xalign = 0;
 
         var secondary_label = new Gtk.Label (content_text);
+        secondary_label.max_width_chars = 50;
+        secondary_label.wrap = true;
         secondary_label.xalign = 0;
 
-        var grid = new Gtk.Grid ();
-        grid.column_spacing = 12;
-        grid.row_spacing = 6;
-        grid.margin_start = grid.margin_end = grid.margin_bottom = 12;
-        grid.attach (image, 0, 0, 1, 2);
-        grid.attach (heading, 1, 0, 1, 1);
-        grid.attach (secondary_label, 1, 1, 1, 1);
+        var cancel = new Gtk.Button.with_label (_("Cancel"));
+
+        var confirm = new Gtk.Button.with_label (button_text);
+        confirm.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+
+        set_default (confirm);
```

Comment:
For some reason I'm seeing the Restart button grab focus here. I actually think Cancel is actually a better choice, anyway.